src/Koffee/MainController.fs

```diff
@@ -43,6 +43,31 @@ let clearSearchProps model =
         Progress = None
     }
 
+let private itemsInDirOrNetHost (fsReader: IFileSystemReader) path history =
+    if path = Path.Network then
+        history.NetHosts
+        |> List.map (sprintf @"\\%s")
+        |> List.choose Path.Parse
+        |> List.map (fun path -> Item.Basic path path.Name NetHost)
+        |> Ok
+    else
+        fsReader.GetItems path |> actionError "open path"
+
+let private historyWithPathAndItsNetHost (history: History) path =
+    let hist = history.WithPath path
+    match path.NetHost with
+    | Some host -> hist.WithNetHost host
+    | None -> hist
+
+let private withBackStackedLocation model path =
+    if path <> model.Location then
+        { model.WithLocation path with
+            BackStack = (model.Location, model.Cursor) :: model.BackStack
+            ForwardStack = []
+            Cursor = 0
```

Comment:
The cursor being set here could be a bit surprising given the name of this function. Please move this line to after 102 with those properties

Reply:
Fully agree. Moved